src/viadot/sources/sap_rfc.py

```diff
@@ -1136,12 +1138,37 @@ def to_df(self, tests: dict | None = None) -> pd.DataFrame:  # noqa: C901, PLR09
                     ):
                         df_tmp = pd.DataFrame(columns=fields)
                         df_tmp[fields] = records
-                        # SAP adds whitespaces to the first extracted column value.
-                        # If whitespace is in unique column, it must be removed to make
-                        # a proper merge.
                         for col in self.rfc_unique_id:
-                            df_tmp[col] = df_tmp[col].str.strip()
-                            df[col] = df[col].str.strip()
+                            # Check in SAP metadata what is the declared
+                            # dtype characters amount
+                            try:
+                                unique_column_len = self._unique_columns_len[col]
+                            except KeyError:
+                                logger.exception(
+                                    f"Missing rfc unique column: {col} length from SAP metadata!"
+                                )
+                                raise
+                            actual_length_of_field = df_tmp[col].str.len()
+                            # Check which rows column values has less characters
+                            # then is defined in SAP data type for each column
+                            rows_missing_whitespaces = (
+                                actual_length_of_field < unique_column_len
+                            )
+                            if any(rows_missing_whitespaces):
+                                # Check how many whitespaces is missing for each row
+                                # column value
+                                logger.info(f"Adding whitespaces for {col} column")
+                                missing_whitespaces_len = (
+                                    unique_column_len - actual_length_of_field
+                                )
+                                df_tmp.loc[rows_missing_whitespaces, col] += (
+                                    np.char.multiply(
+                                        " ",
+                                        missing_whitespaces_len[
+                                            rows_missing_whitespaces
+                                        ],
+                                    )
+                                )
```

Comment:
`np.char.multiply` must stay, otherwise pandas throws errors but I could simplify it a bit by using:

`df_tmp.loc[rows_missing_whitespaces, col] += (
                                    np.char.multiply(
                                        " ",
                                        missing_whitespaces_len
                                    )
                                )`